### DISCUSSION TITLE
Validation functions should be called while submitting the form

### DISCUSSION THREAD

**Discussion by koti5104** [data-driven-forms/react-forms#1272] [Need help] [answered]
Hi Team,

I have written a validate function and passed it to validate function prop in FormRenderer like `<FormRenderer
      validate={validate}`. The validate function is called each time I change any fields value in the form. Can I limit the validation to be called only when I submit the form. 

Thanks in advance.

**Comment by rvsia** [comment, marked as answer]
@koti5104 

Hello, yes. If you want to do a submit validation, you have to just return an object with errors in your submit function, read more [here](https://final-form.org/docs/final-form/types/Config#submission-errors)

**Reply by koti5104** [reply]
@rvsia thank you so much, the approach worked well

**Reply by Nilesh9768** [reply]
Hi @rvsia and @koti5104, 
Could you please provide an fully implemented example for validate function?

Thank you!
